docs/examples/Infra/proxy.md

```diff
@@ -0,0 +1,156 @@
+---
+sidebar_label: Proxy example
+sidebar_position: 7
+---
+
+# Decentralized subscription management and auto-renewal
+
+
+## About
+
+Proxy programs are a well-known design pattern in blockchain development, enabling the delegation of function calls to a separate logic program. This pattern separates business logic from the program's state and address, making it possible to upgrade or modify the logic without impacting the existing state or deployed address.
+
+In this [example](https://github.com/gear-foundation/dapps/tree/master/contracts/proxy-example), we explore a practical implementation of a proxy program that forwards user interactions to a logic program. The proxy manages two main responsibilities:
+
+1. Forwarding state-modifying messages while preserving the original sender's address.
+2. Handling state queries to retrieve data from the logic program.
+This design also supports logic program upgrades, ensuring flexibility and compatibility while maintaining administrative control.
+
+## Program logic
+
+### Message Forwarding
+1. **State-Modifying Messages**
+The `execute_msg` function forwards a state-modifying message to the logic program. It includes the original sender's address (`msg::source()`) in the payload for transparency and accountability.
+```rust
```

Comment:
please specify the local path where the code functions are located
```suggestion
```rust title="proxy-example/proxy/app/src/lib.rs"
```